src/abstractions/src/functions/KernelFunction.ts

```diff
@@ -1,110 +1,94 @@
-import { PromptExecutionSettings } from '../AI';
+import { defaultServiceId } from '../AI';
 import { Kernel } from '../Kernel';
-import { FromSchema, JsonSchema } from '../jsonSchema';
-import { KernelArguments } from './KernelArguments';
-
-export type Fn<Result, Args> = (args: Args, kernel?: Kernel) => Result;
-
-export type FunctionResult<
-  Schema extends JsonSchema | unknown | undefined = unknown,
-  Result = unknown,
-  Args = Schema extends JsonSchema
-    ? FromSchema<Schema>
-    : Schema extends undefined
-      ? undefined
-      : Record<string, unknown>,
-> = {
-  function?: KernelFunction<Schema, Result, Args>;
-  value?: Result;
-  renderedPrompt?: string;
-  metadata?: ReadonlyMap<string, unknown>;
-};
+import { ChatOptions } from '../chatCompletion';
```

Comment:
Is there a reason to keep this if nothing using or is this for future?

Reply:
KernelFunction is a wrapper around AIFunction but it has a different metadata type (has `chatOptions` and `pluginName` in addition to AIFunctionMedata) and it also accepts an instance of `Kernel` in invoke and invokeSteaming